cmd/cosign/cli/verify/verify_blob_test.go

```diff
@@ -1198,8 +1207,8 @@ func TestVerifyBlobCmdInvalidRootCA(t *testing.T) {
 			IgnoreSCT:      true,
 		}
 		err = cmd.Exec(context.Background(), blobPath)
-		if err == nil || !strings.Contains(err.Error(), "certificate signed by unknown authority") {
-			t.Fatalf("expected error with invalid root CA, got %v", err)
+		if err == nil || !strings.Contains(err.Error(), "rekor log public key not found for payload") {
```

Comment:
OH got it -- the test changes the keyless stack / root, and this PR changes the order of verification so that the bundle needed to be verified before the cert chain was verified.

Reply:
Just to ensure the test still makes sense, can you apply this diff?
```
--- a/cmd/cosign/cli/verify/verify_blob_test.go
+++ b/cmd/cosign/cli/verify/verify_blob_test.go
@@ -1173,7 +1173,7 @@ func TestVerifyBlobCmdWithBundle(t *testing.T) {
 func TestVerifyBlobCmdInvalidRootCA(t *testing.T) {
        keyless := newKeylessStack(t)
        // Change the keyless stack.
-       _ = newKeylessStack(t)
+       newKeyless := newKeylessStack(t)
        t.Run("Invalid certificate root when specifying cert via certRef", func(t *testing.T) {
                identity := "hello@foo.com"
                issuer := "issuer"
@@ -1190,8 +1190,8 @@ func TestVerifyBlobCmdInvalidRootCA(t *testing.T) {
 
                // Create bundle
                entry := genRekorEntry(t, hashedrekord.KIND, hashedrekord.New().DefaultVersion(), []byte(blob), leafPemCert, sig)
-               b := createBundle(t, sig, leafPemCert, keyless.rekorLogID, leafCert.NotBefore.Unix()+1, entry)
-               b.Bundle.SignedEntryTimestamp = keyless.rekorSignPayload(t, b.Bundle.Payload)
+               b := createBundle(t, sig, leafPemCert, newKeyless.rekorLogID, leafCert.NotBefore.Unix()+1, entry)
+               b.Bundle.SignedEntryTimestamp = newKeyless.rekorSignPayload(t, b.Bundle.Payload)
                bundlePath := writeBundleFile(t, keyless.td, b, "bundle.json")
                blobPath := writeBlobFile(t, keyless.td, blob, "blob.txt")
                certPath := writeBlobFile(t, keyless.td, string(leafPemCert), "cert.pem")
@@ -1207,8 +1207,8 @@ func TestVerifyBlobCmdInvalidRootCA(t *testing.T) {
                        IgnoreSCT:      true,
                }
                err = cmd.Exec(context.Background(), blobPath)
-               if err == nil || !strings.Contains(err.Error(), "rekor log public key not found for payload") {
-                       t.Fatalf("expected error with rekor public key mismatch, got %v", err)
+               if err == nil || !strings.Contains(err.Error(), "certificate signed by unknown authority") {
+                       t.Fatalf("expected error with certificate signed by unknown authority, got %v", err)
                }
        })
        t.Run("Invalid certificate root when specifying cert in bundle", func(t *testing.T) {
@@ -1227,8 +1227,8 @@ func TestVerifyBlobCmdInvalidRootCA(t *testing.T) {
 
                // Create bundle
                entry := genRekorEntry(t, hashedrekord.KIND, hashedrekord.New().DefaultVersion(), []byte(blob), leafPemCert, sig)
-               b := createBundle(t, sig, leafPemCert, keyless.rekorLogID, leafCert.NotBefore.Unix()+1, entry)
-               b.Bundle.SignedEntryTimestamp = keyless.rekorSignPayload(t, b.Bundle.Payload)
+               b := createBundle(t, sig, leafPemCert, newKeyless.rekorLogID, leafCert.NotBefore.Unix()+1, entry)
+               b.Bundle.SignedEntryTimestamp = newKeyless.rekorSignPayload(t, b.Bundle.Payload)
                bundlePath := writeBundleFile(t, keyless.td, b, "bundle.json")
                blobPath := writeBlobFile(t, keyless.td, blob, "blob.txt")
 
@@ -1243,8 +1243,8 @@ func TestVerifyBlobCmdInvalidRootCA(t *testing.T) {
                        IgnoreSCT:      true,
                }
                err = cmd.Exec(context.Background(), blobPath)
-               if err == nil || !strings.Contains(err.Error(), "rekor log public key not found for payload") {
-                       t.Fatalf("expected error with rekor public key mismatch, got %v", err)
+               if err == nil || !strings.Contains(err.Error(), "certificate signed by unknown authority") {
+                       t.Fatalf("expected error with certificate signed by unknown authority, got %v", err)
                }
        })
 }
```

This uses the correct Rekor keyless stack signer, but just changes the certs, so you can revert the error check change.

Reply:
Done!